pkg/client/liquidation_recovery.go

```diff
@@ -176,17 +186,59 @@ func handleLiquidationRecovery(
 
 // resolveVbyteFee fetches vByte fee for 25 blocks from the bitcoin handle. If a
 // call to Bitcoin API fails the function catches and logs the error but doesn't
-// fail the execution. If a value of vByte fee was not fetched from the bitcoin
-// handle the function tries to read it from a config file. If the value is not
-// defined in the config file it returns a default vByte fee.
-func resolveVbyteFee(bitcoinHandle bitcoin.Handle, tbtcConfig *tbtc.Config) int32 {
+// fail the execution.
+//
+// If a value of vByte fee was returned from the bitcoin handle it is used to
+// calculate transaction fee estimate. If the estimated transaction fee exceeds
+// the transaction value more than 5%, then the lesser of the suggested vByte fee
+// or configured MaxFeePerVByte is used.
+//
+// If a value of vByte fee was not fetched from the bitcoin handle the function
+// tries to read it from a config file. If the value is not defined in the config file
+// it returns a default vByte fee.
+func resolveVbyteFee(
+	bitcoinHandle bitcoin.Handle,
+	tbtcConfig *tbtc.Config,
+	previousOutputValue int32,
+) int32 {
 	vbyteFee, vbyteFeeError := bitcoinHandle.VbyteFeeFor25Blocks()
 	if vbyteFeeError != nil {
 		logger.Errorf(
 			"failed to retrieve a vbyte fee estimate: [%v]",
 			vbyteFeeError,
 		)
-		// Since the electrs connection is optional, we don't return the error
+		// Since the electrs connection is optional, we don't return the error.
+	}
+	if vbyteFee > 0 {
+		// Fee computation requires that the full transaction be assembled. The
+		// transaction fee should be estimated by assuming the final transaction
+		// will be 175 vBytes, which should be very close to accurate given the
+		// regular structure of liquidation refund transactions.
+		estimatedTransactionFee := vbyteFee * estimatedTransactionSizeVByte
+
+		fivePercentPreviousOutput := previousOutputValue / 20 // 5% of UTXO
```

Comment:
Do we lose precision here because everything is integers? Does that actually matter? 🤔

Reply:
We do lose precision, but I believe it doesn't matter to us that much here.

We expect `previousOutputValue` to be relatively big values. The lowest TBTC deposit lot size is 0.01 BTC, so `previousOutputValue` will be `1000000` (0.01 BTC -> satoshis).

We will use the calculated value in `estimatedTransactionFee > fivePercentPreviousOutput` comparison.
`estimatedTransactionFee` will be min. `175` (`estimatedTransactionFee = vbyteFee * estimatedTransactionSizeVByte`)

The lowest possible scenario where precision would matter is `estimatedTransactionFee` to be `175` or `176`. Which corresponds to `previousOutputValue` of `175*20 = 3500` and `176*20=3520`.

Also, the calculation result is rounded down so it works in the operator's favor in `estimatedTransactionFee > fivePercentPreviousOutput` check.

Please let me know if I made some mistakes in my assumptions and we should be more precision caucious.

Reply:
Okay! Double-checked that math there and it all looks good. Thanks for proving that out for me!